File: leohoot/src/app/model/quiz.ts
import { QuestionTeacher } from './question-teacher';
import { Tag } from './tag';

export interface Quiz {
  title: string;
  description: string;
  questions: QuestionTeacher[];
  tags: Tag[];
  creator: string;
  imageName: string;
  isPublic: boolean;
  isFavorited: boolean;
  id?: number;
}

File: leohoot-backend/Hubs/Quiz.cs
using System;

public class Quiz
{
    public string Title { get; }
    public string Description { get; }
    public User Creator { get; }
    public List<Question> Questions { get; }

    public Quiz(string title, string description, User creator, List<Question> questions)
    {
        this.Title = title;
        this.Description = description;
        this.Creator = creator;
        this.Questions = questions;
    }
}